pkg/skaffold/docker/remote.go

```diff
@@ -84,7 +86,7 @@ func retrieveRemoteConfig(identifier string) (*v1.ConfigFile, error) {
 }
 
 func remoteImage(identifier string) (v1.Image, error) {
-	ref, err := name.ParseReference(identifier, name.WeakValidation)
+	ref, err := name.ParseReference(identifier, name.StrictValidation)
```

Comment:
This might have unintended consequences since it's used in a few places. If you're confident in your test coverage, it's probably fine 😃

Reply:
yeah it's slightly risky, but it's the "right" thing to do since we should never be passing an invalid reference anyway. I'll take responsibility if it breaks something :eyes: